x/datapool/types/message_pool.go

```diff
@@ -0,0 +1,168 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+var _ sdk.Msg = &MsgRegisterDataValidator{}
+
+func NewMsgRegisterDataValidator(address string, dataValidator *DataValidator) *MsgRegisterDataValidator {
+	return &MsgRegisterDataValidator{
+		Address:         address,
+		ValidatorDetail: dataValidator,
+	}
+}
+
+func (msg *MsgRegisterDataValidator) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgRegisterDataValidator) Type() string {
+	return "RegisterDataValidator"
+}
+
+func (msg *MsgRegisterDataValidator) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Address)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
+	}
+	return nil
+}
+
+func (msg *MsgRegisterDataValidator) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgRegisterDataValidator) GetSigners() []sdk.AccAddress {
+	dataValidator, err := sdk.AccAddressFromBech32(msg.Address)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{dataValidator}
+}
+
+var _ sdk.Msg = &MsgCreatePool{}
```

Comment:
Because I need to write test code, I implemented all of our transaction such as `MsgCreatePool`, `MsgSellData`, `MsgBuyDataAccessNFT` in `message_pool.go`.